draft-ietf-taps-interface.md

```diff
@@ -521,18 +522,54 @@ strict requirement, as it signifies a reduction in reliability. The
+### Interface Instance or Type {#prop-interface}
 
 Type: Tuple (Enumeration, Preference)
 
-This property specifies which kind of access network interface,
-e.g., WiFi, Ethernet, or LTE, to prefer over others for this Connection, in
-case they are available. In general, Interface Types should be used only with
-the `Prefer` and `Prohibit` preference level. Specifically, using the
-`Require` preference level for Interface Type may limit path selection in a
-way that is detrimental to connectivity. The default is to use the default
-interface configured in the system policy.
-The valid values for the access network interface kinds are implementation specific.
+This property allows the application to select which specific network interfaces
+or categories of interfaces it wants to `Require`, `Prohibit`, or `Prefer`.
+
+If a system supports discovery of specific interface identifiers, such as `en0` or `eth0`
+on Unix-style systems, an implemention should allow using these identifiers to
+define path preferences. Note that marking a specific interface as `Required`
+strictly limits path selection to a single interface, and leads to less flexible and
+resilient connection establishment.
+
+The set of valid interface types is implementation- and system-specific. For example,
+on a mobile device, there may be `Wi-Fi` and `Cellular` interface types available;
+whereas on a desktop computer, there may be `Wi-Fi` and `Wired Ethernet`
+interface types available. Implementations should provide all types that are supported
+on some system to all systems, in order to allow applications to write generic code.
+For example, if a single implementation is used on both mobile devices and desktop
+devices, it should define the `Cellular` interface type for both systems, since an
+application may want to always `Prohibit Cellular`. Note that marking a specific 
+interface type as `Required` limits path selection to a small set of interfaces, and leads
+to less flexible and resilient connection establishment.
+
+The set of interface types is expected to change over time as new access technologies
+become available.
+
+### Provisioning Domain Instance or Type {#prop-pvd}
+
+Type: Tuple (Enumeration, Preference)
+
+Similar to interface instances and types {{prop-interface}}, this property allows
+the application to control path selection by selecting which specific Provisioning Domains
+or categories of Provisioning Domains it wants to  `Require`, `Prohibit`, or `Prefer`.
+Provisioning Domains define consistent sets of network properties that may be more
+specific than network interfaces {{RFC7556}}.
+
+The indentification of a specific Provisioning Domain (PvD) is defined to be implementation-
+and system-specific. For example, this may be a string name or an integer. As with 
+requiring specific interfaces, requiring a specific PvD strictly limits path selection.
+
+Categories or types of PvDs are also defined to be implementation- and system-specific.
```

Comment:
- We should have at least a portable way to specify PvD strings. 
- Maybe we should also have a portable way to map 3GPP APNs tp PvDs (but most probably out of scope for this document)
Other stuff can indeed be implementation specific.

Reply:
There is no portable definition of a PvD string, and we certainly aren't going to go into the business of creating APN registries...

Reply:
Oh snap… RFC7558 really does not define any format… If we take the assumption that PvDs are FQDNs from draft-bruneau-pvd, it would be so much easier, as we could be sure the PvD does not contain a colon.
This would enable clearly sepatating explicit PvDs (without colon) from PvD classes (with colon)

Reverring to a specific APN could be realized with a APN class: `APN:[apn name]`

Reply:
This document is certainly not the place to go into the debate of how to identify PvDs. This is a very long and complicated discussion, and we already have two platforms (Android and iOS) that handle them with quite different identifiers. Thus, the text here represents the reality that these will indeed need to be system-specific.

Reply:
Added text to explain why the PvD identifier isn't strongly defined.